scripts/pylib/twister/twisterlib.py

```diff
@@ -2017,6 +2017,7 @@ def run_cmake(self, args=[]):
         cmake_args = [
             f'-B{self.build_dir}',
             f'-S{self.source_dir}',
+            f'-DEMU_PLATFORM={self.platform.simulation}',
```

Comment:
If you add this CMake argument `EMU_PLATFORM` to every CMake build running via Twister it enable to run simulator properly to this set in yaml file, as you said in previously comments. But this set CMake argument cause also unexpected CMake warning log in situation, when you run Twister on real target (not on emulaiton). In this case `self.platform.simulation` variable is set as "na" and this cause appear in CMake log misleading information: `CMake file for na not found!`. I propose setting this CMake arguments by `extra_args` in yaml file or eventually (but not recommended) by create if statement in `twisterlib.py` and add this CMake argument only if `self.platform.simulation` is not "na"

Reply:
If I understand it correctly, platform schema doesn't define `extra_args` field, so it's not possible for it to be set for each board individually. Or did you mean the `extra_args` as defined by testcase schema? If so, that would mean forcing a testcase to be run on one emulation platform regardless of the board that is being emulated.

Reply:
You're right, `extra_args` concern only testcase schema - sorry, I omit it. 
So you can consider add `EMU_PLATFORM` in conditional statement like this:
```
if self.platform.simulation != "na":
    cmake_args.append(f'-DEMU_PLATFORM={self.platform.simulation}')
```

Or try change somehow if conditional statement in `CMakeLists.txt` to avoid misleading warning in case of use Twister on real board from there: [CMakeLists.txt](https://github.com/zephyrproject-rtos/zephyr/blob/af7dad518b9902fdb48d32e350c0b920c91192cd/CMakeLists.txt#L1602)

Reply:
I've added the conditional statement you suggested. Thanks for spotting the potential problem with this change!